modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/XmlExtensionParserDelegate.java

```diff
@@ -156,7 +158,25 @@ ValueResolver parseParameter(ElementDescriptor element, ParameterModel parameter
             return new StaticValueResolver<>(getAttributeValue(element, parameterModel.getName(), parameterModel.getDefaultValue()));
         }
 
-        return parseElement(element, parameterModel.getName(), parameterModel.getType(), parameterModel.getDefaultValue());
+        Optional<SubtypesModelProperty> subtypesProperty = parameterModel.getModelProperty(SubtypesModelProperty.class);
+        if (subtypesProperty.isPresent())
+        {
+            List<Class<?>> subtypes = subtypesProperty.get().getSubtypes();
+            Optional<Class<?>> subtypeChildElement = subtypes.stream()
+                    .filter(s -> element.getChildByName(hyphenize(s.getSimpleName(), "as", parameterModel.getName())) != null)
```

Comment:
what is this "as" thing? Sounds very very weird.... best case put it in a constant